examples/requirements/docstring.py

```diff
@@ -0,0 +1,102 @@
+
+from avocado import Test
+from avocado.utils import vmimage
+
+
+class GPL2(Test):
+    """
+    Example test using docstring requirements.
+
+    :avocado: requirement={"type": "file", "name": "gpl-2.0.txt", "locations": "https://mirrors.kernel.org/gnu/Licenses/gpl-2.0.txt", "asset_hash": "4cc77b90af91e615a64ae04893fdffa7939db84c"}
```

Comment:
I'm not getting this requirement (or any other `file` type requirement) fulfilled:

```
$ avocado-requirements examples/requirements/docstring.py 
Fulfilling image requirement: cirros
 Image requirement: /home/cleber/avocado/data/cache/by_location/e3507e8a76e35c36e4542bac145441d9b7f61fbd/cirros-0.5.0-x86_64-disk.img
```

And then the test cancels (as it should because of `cancel_on_missing=True`):

```
JOB ID     : b4b995946a97c41a812263ae35df2f4e045800dc
JOB LOG    : /home/cleber/avocado/job-results/job-2020-05-06T15.42-b4b9959/job.log
 (1/5) examples/requirements/docstring.py:GPL2.test_find_gpl: CANCEL: Missing asset gpl-2.0.txt (0.00 s)
 (2/5) examples/requirements/docstring.py:GPL3.test_find_gpl: CANCEL: Missing asset gpl-3.0.txt (0.00 s)
 (3/5) examples/requirements/docstring.py:LGPL2.test_find_lgpl2: CANCEL: Missing asset lgpl-2.0.txt (0.01 s)
 (4/5) examples/requirements/docstring.py:LGPL2.test_find_lgpl2_1: CANCEL: Missing asset lgpl-2.0.txt (0.01 s)
 (5/5) examples/requirements/docstring.py:Vmimage.test_find_image: PASS (0.72 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 4
JOB TIME   : 1.12 s
JOB HTML   : /home/cleber/avocado/job-results/job-2020-05-06T15.42-b4b9959/results.html
```

Any ideas?

Reply:
Seems it is missing a commit on your side. The files are supposed to go to `by_name` directory, but I see yours are going to `by_location`. I just cleaned up my Avocado cache and ran again:

Ran the test, so it fails:

```
[wrampazz@wrampazz avocado.dev]$ avocado run examples/requirements/docstring.py 
/home/wrampazz/src/avocado/avocado.dev/avocado/plugins/run.py:284: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast and keep-tmp. 
  warnings.warn("The following arguments will be changed to boolean soon: "
JOB ID     : 9f854279295acc68db8688be9eb6b8a530d6737d
JOB LOG    : /home/wrampazz/avocado/job-results/job-2020-05-06T16.45-9f85427/job.log
 (1/5) examples/requirements/docstring.py:GPL2.test_find_gpl: CANCEL: Missing asset gpl-2.0.txt (0.01 s)
 (2/5) examples/requirements/docstring.py:GPL3.test_find_gpl: CANCEL: Missing asset gpl-3.0.txt (0.01 s)
 (3/5) examples/requirements/docstring.py:LGPL2.test_find_lgpl2: CANCEL: Missing asset lgpl-2.0.txt (0.00 s)
 (4/5) examples/requirements/docstring.py:LGPL2.test_find_lgpl2_1: CANCEL: Missing asset lgpl-2.0.txt (0.00 s)
 (5/5) examples/requirements/docstring.py:Vmimage.test_find_image: PASS (2.15 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 4
JOB TIME   : 2.57 s
JOB HTML   : /home/wrampazz/avocado/job-results/job-2020-05-06T16.45-9f85427/results.html
```

Ran the resolver:

```
[wrampazz@wrampazz avocado.dev]$ avocado-requirements examples/requirements/docstring.py 
Fulfilling file requirement: gpl-2.0.txt
 File requirement: /home/wrampazz/avocado/data/cache/by_name/gpl-2.0.txt
Fulfilling file requirement: gpl-3.0.txt
 File requirement: /home/wrampazz/avocado/data/cache/by_name/gpl-3.0.txt
Fulfilling file requirement: lgpl-2.0.txt
 File requirement: /home/wrampazz/avocado/data/cache/by_name/lgpl-2.0.txt
Fulfilling file requirement: lgpl-2.1.txt
 File requirement: /home/wrampazz/avocado/data/cache/by_name/lgpl-2.1.txt
Fulfilling image requirement: cirros
 Image requirement: /home/wrampazz/avocado/data/cache/by_location/e3507e8a76e35c36e4542bac145441d9b7f61fbd/cirros-0.5.0-x86_64-disk.img
```

Then ran again the test:

```
[wrampazz@wrampazz avocado.dev]$ avocado run examples/requirements/docstring.py 
/home/wrampazz/src/avocado/avocado.dev/avocado/plugins/run.py:284: FutureWarning: The following arguments will be changed to boolean soon: sysinfo, output-check, failfast and keep-tmp. 
  warnings.warn("The following arguments will be changed to boolean soon: "
JOB ID     : 87c01266c8cc6a22e7613bd447dcba45459c0480
JOB LOG    : /home/wrampazz/avocado/job-results/job-2020-05-06T16.45-87c0126/job.log
 (1/5) examples/requirements/docstring.py:GPL2.test_find_gpl: PASS (0.03 s)
 (2/5) examples/requirements/docstring.py:GPL3.test_find_gpl: PASS (0.03 s)
 (3/5) examples/requirements/docstring.py:LGPL2.test_find_lgpl2: PASS (0.05 s)
 (4/5) examples/requirements/docstring.py:LGPL2.test_find_lgpl2_1: PASS (0.02 s)
 (5/5) examples/requirements/docstring.py:Vmimage.test_find_image: PASS (2.24 s)
RESULTS    : PASS 5 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 2.80 s
JOB HTML   : /home/wrampazz/avocado/job-results/job-2020-05-06T16.45-87c0126/results.html
[wrampazz@wrampazz avocado.dev]$ 
```

Reply:
@clebergnu ah, check my first comment in this PR: It depends on #3748.

Reply:
Oh, I get it. I'm running this exact commit, 41ce62e1c8835acfb452274f7f7e2972dcd4d060.  You may want to rebase this (or not ;).

Reply:
Yep, rebased and it works as expected. Thanks.

Reply:
I just pushed the commit rebased. It should be good now.